python/nano/src/bigdl/nano/pytorch/trainer/Trainer.py

```diff
@@ -104,3 +108,24 @@ def __init__(self, num_processes: int = 1,
 
             super().__init__(accelerator=accelerator,
                              plugins=[plugin], *args, **kwargs)
+
+    @staticmethod
+    def compile(model: nn.Module, loss: _Loss = None, optimizer: torch.optim = None):
+        """
+        Compile a pytorch model into a pytorch-lightning model and return it.
+        Args:
+            model: A pytorch model instance.
+            loss: Loss for LightningModule.
+            optimizer: Optimizer for LightningModule.
+
+        Returns: A LightningModule converted from model.
+
+        """
+        if isinstance(model, pl.LightningModule):
+            if loss:
+                model.loss = loss
+            if optimizer:
+                model.optimizer = optimizer
+            return model
```

Comment:
Why do we want to support this? Can we guarantee this is correct if `configure_optimizers` returns a different optimizer?

Reply:
Correct. Added assertion to check if the inputs are correct.